src/ResourceFormatter.php

```diff
@@ -77,6 +78,51 @@ public static function encode( $id, $data ) {
 		);
 	}
 
+	/**
+	 * @param array $printRequests
+	 * @param array $ask
+	 * @return array
+	 */
+	private static function appendPreferredPropertyLabel( $printRequests, $ask ) {
```

Comment:
Better to use types in PHP and remove the doc block

Reply:
Tho you could keep the doc block and add more detailed type info such as `PrintRequest[]` or whatever the class name actually is.

Reply:
I've added the param type
` * @param PrintRequest[] $printRequests`

Reply:
Without it being imported, this references to SRF\PrintRequest, which does not exist. 

You need to make sure the type is actually correct. Otherwise it does not help by providing autocompletion and error detection in the IDE. PhpStorm can autocomplete on the type in the comment and even import it automatically for you.

Reply:
so it should rather be @param `SMW\Query\PrintRequest[] $printRequests` ?

Reply:
That would resolve to `SRF\SMW\Query\PrintRequest[]`. You need a backslash at the front: `\SMW\Query\PrintRequest[]`

Though better is to import the name with an import statement, so you do not have the FQN in the docs.

Reply:
I have added the import statement